src/main.rs

```diff
@@ -33,29 +37,33 @@ impl Terminal {
         }
     }
 
-    fn ask_for_new_todo(&mut self) -> Todo {
+    fn ask_for_new_todo(&mut self) -> Result<Todo, Error> {
         writeln!(self.stdout, "\nQuer adicionar um novo TODO 📝?\ndigite (sim) para confirmar 👍  ou (nao) para negar 👎").unwrap();
-        
+
         let mut resposta = input();
 
-        self.stdin.read_line(&mut resposta).unwrap();
+        self.stdin.read_line(&mut resposta)?;
 
         if resposta.trim() == "sim" {
-            writeln!(self.stdout, "\nQual TODO 📝 deseja criar?").unwrap();
-            
+            writeln!(self.stdout, "\nQual TODO 📝 deseja criar?")?;
+
             let mut novo_todo = input();
 
-            self.stdin.read_line(&mut novo_todo).unwrap();
+            self.stdin.read_line(&mut novo_todo)?;
 
-            Todo { message: novo_todo }
+            Ok(Todo { message: novo_todo })
         } else {
-            writeln!(self.stdout, "\nAté a próxima 👋 e volte sempre!🫶\n").unwrap();
-            
+            writeln!(self.stdout, "\nAté a próxima 👋 e volte sempre!🫶\n")?;
+
             std::process::exit(0);
         }
     }
 
     fn show_todo(&mut self, todo: &Todo) {
```

Comment:
O método `show_todo` precisa também do tratamento de erros.
```suggestion
    fn show_todo(&mut self, todo: &Todo) ->  Result<(), Error> {
```